pkg/controller/machine/machine_controller.go

```diff
@@ -174,11 +146,9 @@ func Add(
 		kubeconfigProvider:               kubeconfigProvider,
 		providerData:                     providerData,
 		joinClusterTimeout:               joinClusterTimeout,
-		externalCloudProvider:            externalCloudProvider,
 		name:                             name,
 		bootstrapTokenServiceAccountName: bootstrapTokenServiceAccountName,
 		skipEvictionAfter:                skipEvictionAfter,
-		nodeSettings:                     nodeSettings,
```

Comment:
keep for backward compatibility. e.g. machine created by old `machineset_controller` and processed by new `machine_controller`